aws-cs-fargate/README.md

```diff
@@ -0,0 +1,124 @@
+    ```
+
+3. Deploy everything with a single `pulumi up` command. This will show you a preview of changes first, which
+   includes all of the required AWS resources (clusters, services, and the like). Don't worry if it's more than
+   you expected -- this is one of the benefits of Pulumi, it configures everything so that so you don't need to!
+
+    ```bash
+    $ pulumi up
+    ```
+
+    After being prompted and selecting "yes", your deployment will begin. It'll complete in a few minutes:
+
+    ```
+    Updating (dev):
+         Type                                        Name                Status
+     +   pulumi:pulumi:Stack                         aws-cs-fargate-dev  created
+     +   ├─ aws:ec2:SecurityGroup                    web-sg              created
+     +   ├─ aws:ecs:Cluster                          app-cluster         created
+     +   ├─ aws:iam:Role                             task-exec-role      created
+     +   ├─ aws:elasticloadbalancingv2:TargetGroup   web-tg              created
+     +   ├─ aws:ecr:Repository                       app-repo            created
+     +   ├─ docker:image:Image                       app-img             created
+     +   ├─ aws:iam:RolePolicyAttachment             task-exec-policy    created
+     +   ├─ aws:ecs:TaskDefinition                   app-task            created
+     +   ├─ aws:elasticloadbalancingv2:LoadBalancer  web-lb              created
+     +   └─ aws:ecs:Service                          app-svc             created
+
+    Outputs:
+        url: "http://web-lb-23139b7-1806442625.us-east-1.elb.amazonaws.com"
+
+    Resources:
+        + 11 created
+
+    Duration: 3m41s
+
+    Permalink: https://app.pulumi.com/acmecorp/aws-cs-fargate/dev/updates/1
+    ```
+
+   Notice that the automatically assigned load-balancer URL is printed as a stack output.
+
+4. At this point, your app is running -- let's curl it. The CLI makes it easy to grab the URL:
+
+    ```bash
+    $ curl $(pulumi stack output url)
+    Hello World!
+    ```
+
+5. Try making some changes and rerunning `pulumi up`.
+
+   If you just change the application code, and deploy the results, for example, only the Docker image
+   will be updated and rolled out. Try changing `"Hello World!"` inside of `App/Startup.cs` to `"Hello Pulumi!"`:
+
+   ```bash
+   $ pulumi up
+   Updating (dev):
+         Type                       Name                Plan        Info
+         pulumi:pulumi:Stack        aws-cs-fargate-dev
+     +-  ├─ aws:ecs:TaskDefinition  app-task            replaced    [diff: ~containerDefinitions]
+     ~   ├─ aws:ecs:Service         app-svc             updated     [diff: ~taskDefinition]
+         └─ docker:image:Image      app-img
+
+    Resources:
+        ~ 1 updated
+        +-1 replaced
+        2 changes. 9 unchanged
+   ```
+
+   Notice that `pulumi up` redeploys just the parts of the application/infrastructure that you've edited.
```

Comment:
So does the re-publishing of the docker image work?

Reply:
It does! I filed an issue for the lack of diffs (https://github.com/pulumi/pulumi-docker/issues/131) which led me astray. The root problem was that I was using the ECR registry URL and _not_ the built image's full name, which is just the ECR registry plus the hash. The hash is required to ensure the ECS service gets restarted upon a new image being pushed.